src/pages/email-change-request/[token].tsx

```diff
@@ -1,19 +1,81 @@
 import * as React from 'react'
 import LoginRequired from 'components/login-required'
-import axios from 'axios'
 import {useRouter} from 'next/router'
 import {useViewer} from 'context/viewer-context'
 import toast from 'react-hot-toast'
+import axios from 'axios'
+import isEmpty from 'lodash/isEmpty'
+import cx from 'classnames'
+import {
+  getAuthorizationHeader,
+  getTokenFromCookieHeaders,
+  AUTH_DOMAIN,
+} from 'utils/auth'
 
 async function confirmEmailChangeRequest(token: any) {
-  const {data} = await axios.get(
+  const {data} = await axios.patch(
     `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1/email_change_requests/${token}`,
+    {},
+    {
+      headers: {...getAuthorizationHeader()},
+    },
   )
 
   return data
 }
 
-const EmailChangeRequest: React.FunctionComponent = () => {
+type emailChangeRequestData = {
+  new_email: string
+  current_email: string
+}
+
+export async function getServerSideProps(context: any) {
```

Comment:
Request the current/new email addresses from the server in `getServerSideProps`. If the token is expired, used, or unknown, we display an error message with options.